cmd/axelard/cmd/vald/start.go

```diff
@@ -377,10 +379,28 @@ func createTSSMgr(broadcaster broadcasterTypes.Broadcaster, cliCtx client.Contex
 		gg20client := tofnd.NewGG20Client(conn)
 		multiSigClient := tofnd.NewMultisigClient(conn)
 
-		tssMgr := tss.NewMgr(gg20client, multiSigClient, cliCtx, 2*time.Hour, valAddr, broadcaster, logger, cdc)
+		// retrieve key id mappings
+
+		queryClient := tssTypes.NewQueryServiceClient(cliCtx)
+		res, err := queryClient.ValidatorKey(ctx, &tssTypes.ValidatorKeyRequest{
+			Address: valAddr,
+		})
+
+		status, _ := status.FromError(err)
+
+		switch status.Code() {
+		case codes.OK:
+		case codes.NotFound:
+			return nil, fmt.Errorf("query result was not found")
+		default:
+			return nil, sdkerrors.Wrap(err, "failed to execute query")
+		}
+
+		tssMgr := tss.NewMgr(gg20client, multiSigClient, cliCtx, 2*time.Hour, valAddr, res, broadcaster, logger, cdc)
```

Comment:
```suggestion
		tssMgr := tss.NewMgr(gg20client, multiSigClient, cliCtx, 2*time.Hour, valAddr, res.Keys, broadcaster, logger, cdc)
```
The manager should not depend on the query-specific response but generic data within.

Reply:
Yeah, I kept it since there was another query specific type inside it, but now I explicitly convert it to a simple map.